src/features/CareViewPatientsSummary/components/CareViewPatientsSummary.jsx

```diff
@@ -1,51 +1,200 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
 import propTypes from "prop-types";
 import "../styles/CareViewPatientsSummary.scss";
 import { HospitalBed16 } from "@carbon/icons-react";
 import { Link } from "carbon-components-react";
 import { FormattedMessage } from "react-intl";
+import { getSlotsForPatients } from "../../CareViewSummary/utils/CareViewSummary";
+import Clock from "../../../icons/clock.svg";
+import {
+  epochTo24HourTimeFormat,
+  getPreviousNearbyHourEpoch,
+  getTimeInFuture,
+} from "../../../utils/DateTimeUtils";
+import { getAdministrationStatus } from "../../../utils/CommonUtils";
+import SVGIcon from "../../SVGIcon/SVGIcon";
+import { CareViewContext } from "../../../context/CareViewContext";
 
 export const CareViewPatientsSummary = (props) => {
+  const [slotDetails, setSlotDetails] = useState([]);
+  const { careViewConfig, ipdConfig } = useContext(CareViewContext);
   const { patientsSummary } = props;
-  return (
-    <table className={"care-view-patient-table"}>
-      <tbody>
-        {patientsSummary.map((patientSummary, idx) => {
-          const { patientDetails, bedDetails } = patientSummary;
-          const { person } = patientDetails;
+  const timeframeLimitInHours = careViewConfig.timeframeLimitInHours;
+  const currentEpoch = Math.floor(new Date().getTime() / 1000);
+  const nearestHourEpoch = getPreviousNearbyHourEpoch(currentEpoch);
+
+  const fetchSlots = async (patients) => {
+    const patientUuids = patients.map((patient) => patient.patientDetails.uuid);
+    const currentTime = getPreviousNearbyHourEpoch(
+      Math.floor(Date.now() / 1000)
+    );
+    const endTime = getTimeInFuture(currentTime, timeframeLimitInHours);
+    const response = await getSlotsForPatients(
+      patientUuids,
+      currentTime,
+      endTime
+    );
+    setSlotDetails(response);
+  };
+
+  useEffect(() => {
+    if (patientsSummary.length > 0) {
+      fetchSlots(patientsSummary);
+    }
+  }, [patientsSummary]);
+
+  const renderStatusIcon = (slot) => {
```

Comment:
Can we move this into a new file src/features/CareViewPatientsSummary/utils/CareViewPatientsSummary.jsx